cmd/simd/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/libs/cli"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func main() {
+
+	cobra.EnableCommandSorting = true
+
+	// Read in the configuration file for the sdk
+	config := sdk.GetConfig()
+	config.SetBech32PrefixForAccount(sdk.Bech32PrefixAccAddr, sdk.Bech32PrefixAccPub)
+	config.SetBech32PrefixForValidator(sdk.Bech32PrefixValAddr, sdk.Bech32PrefixValPub)
+	config.SetBech32PrefixForConsensusNode(sdk.Bech32PrefixConsAddr, sdk.Bech32PrefixConsPub)
+	config.Seal()
+
+	rootComd := &cobra.Command{
+		Use:   "simd",
+		Short: "simulation app",
```

Comment:
Try setting  the following here and removing from the child commands:

```go
	rootClientCmd.PersistentPreRunE = func(cmd *cobra.Command, _ []string) error {
		return client.SetCmdClientContextHandler(initClientCtx, cmd)
	}
```

Reply:
No, It won't work. Same issue with integration test.

Reply:
The error in  `TestCLICreateValidator`: 
```
gobash.go:60: Running /home/runner/work/cosmos-sdk/cosmos-sdk/build/simd tx send --keyring-backend=test foo cosmos16k7m4afw43ppeja3qzr75en6l4a0emw7pmvefg 10000000stake --home=/tmp/sdk_integration_TestCLICreateValidator_807107126/.simcli --node=tcp://0.0.0.0:38269 --chain-id=test-chain-U1bavE --output=json -y
    executors.go:41: Stderr: panic: runtime error: invalid memory address or nil pointer dereference
        [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xfe56e5]
```

is caused by this line of code:

https://github.com/dauTT/cosmos-sdk/blob/single-binary-6419/client/tx/tx.go#L290

![Selection_093](https://user-images.githubusercontent.com/30392990/86415635-f0a3c580-bcc7-11ea-8317-0b53a188ecb5.png)

I am not sure exactly the reason for this.

Beside that, I have tried to run the same test case with the master branch but it  is failing with a different error message.

Reply:
Ok I'll pull down the branch and test it myself. Thanks!